### PR TITLE
usb: msc: doc: Add proper Doxygen documentation

### PR DESCRIPTION
**Pull request by kartben**
Fixed several improperly tagged "groups", and added missing documentation.

### CODE REVIEW

**Review by tmon-nordic — APPROVED** [review]

File: include/zephyr/usb/class/usb_msc.h [hunk 1]
```diff
@@ -46,64 +46,139 @@
...
 #define ZEPHYR_INCLUDE_USB_CLASS_USB_MSC_H_
 
-/** MSC Subclass and Protocol Codes */
-#define SCSI_TRANSPARENT_SUBCLASS	0x06
-#define BULK_ONLY_TRANSPORT_PROTOCOL	0x50
+/**
+ * @name MSC Subclass and Protocol Codes
+ * @{
+ */
+#define SCSI_TRANSPARENT_SUBCLASS	0x06	/**< SCSI transparent command set sub-class */
```
Comment (jfischer-no) [inline include/zephyr/usb/class/usb_msc.h:53]: Can we use the same style in the file? (This also better shows how unnecessary these comments are.)
```
/** SCSI transparent command set sub-class */
#define SCSI_TRANSPARENT_SUBCLASS	0x06
```

**Review by jfischer-no — CHANGES_REQUESTED** [review]
This is an internal API and should probably be hidden internally and not use any doxygen comments.

File: include/zephyr/usb/class/usb_msc.h (hunk 1 above)
Comment (kartben) [reply include/zephyr/usb/class/usb_msc.h:53]: the thinking was to not unecessarily space out the various defines by having the comments on the same line. All defines have a consistent style across the file, I think?

**Comment by kartben** [comment]
> This is an internal API and should probably be hidden internally and not use any doxygen comments.

Looks like it has been taken care of as part of aecb1ca2e7 - thx!

**Pull request closed without merging**